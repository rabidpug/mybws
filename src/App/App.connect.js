import $ from 'redux-methods';
import { connect, } from 'react-redux';
const { changePublic, listenPublic, } = $.user;
const { isOnline, isSidebarCollapsed, } = $.ui;
const dispatch = {
  changePublic,
  isOnline      : isOnline.set,
  listenPublic,
  setModal      : $.ui.modal.set,
  // setWidth      : $.browser.set,
  toggleSidebar : isSidebarCollapsed.toggle,
};
const state = state => ( {
  activeRole         : $.user.profile.getActiveRole( state ),
  activeStore        : $.user.profile.getActiveStore( state ),
  isAuthenticated    : state.user.isAuthenticated,
  isSidebarCollapsed : state.ui.isSidebarCollapsed,
  isSmallDisplay     : state.browser < 768,
  isSocketConnected  : state.user.socket,
  isSocketConnecting : state.inProgress.connectingSocket,
  isUserOnline       : state.ui.isOnline,
  modal              : state.ui.modal,
  profile            : state.user.profile,
} );
const connectApp = connect( state,
                            dispatch );

export default connectApp;
