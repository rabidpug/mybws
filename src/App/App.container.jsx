import 'react-toastify/dist/ReactToastify.css'
import 'react-tippy/dist/tippy.css'

import { CSSTransition, TransitionGroup, } from 'react-transition-group'
import React, { PureComponent, } from 'react'
import ContentSwitch from './components/ContentSwitch'
import GlobalToast from 'Common/components/GlobalToast'
import Layout from 'Common/components/Layout'
import LoggingIn from './components/LoggingIn'
import SwipeMask from 'Common/components/SwipeMask'
import actionMenu from './helpers/actionMenu'
import fadeTransition from 'Common/styles/fadeTransition.scss'
import getQueryVariable from 'Common/helpers/getQueryVariable'
import gqlApp from './App.gql'
import { hot, } from 'react-hot-loader'
import iconLibrary from './helpers/iconLibrary'
import navMenu from './helpers/navMenu'
import { renderRoutes, } from 'react-router-config'
import { withRouter, } from 'react-router-dom'

iconLibrary()

@hot( module )
@withRouter
@gqlApp
export default class App extends PureComponent {
  state = { swipeWidth: 0, }

  componentDidMount () {
    const { updateBrowser, updateIsOnline, loaded, } = this.props

    loaded && this.handleAuthRedirect()

    window.addEventListener( 'beforeinstallprompt', this.handleAddToHome )

    updateIsOnline()

    window.addEventListener( 'online', updateIsOnline )

    window.addEventListener( 'offline', updateIsOnline )

    window.addEventListener( 'resize', updateBrowser )
  }

  componentDidUpdate () {
    getQueryVariable( 'token', location ) && this.handleAuthRedirect()
  }

  componentWillUnmount () {
    const { updateBrowser, updateIsOnline, } = this.props

    window.removeEventListener( 'beforeinstallprompt', this.handleAddToHome )

    window.removeEventListener( 'online', updateIsOnline )

    window.removeEventListener( 'offline', updateIsOnline )

    window.removeEventListener( 'resize', updateBrowser )
  }

  handleAuthRedirect = () => {
    const { login, loaded, } = this.props

    const JWT = getQueryVariable( 'token', location )

    const refreshToken = getQueryVariable( 'refreshToken', location )

    if ( JWT || refreshToken && loaded ) {
      login( {
        variables: {
          JWT,
          refreshToken,
        },
      } )
    }
  }

  handleAddToHome = e => {
    e.preventDefault()

    e.prompt()
  }

  onSwipingRight = ( e, swipeWidth ) => {
    const { data: { ui: { isSidebarCollapsed, }, }, } = this.props

    if ( isSidebarCollapsed ) this.setState( { swipeWidth, } )
  }

  onSwiped = (
    e, deltaX, deltaY, isFlick
  ) => {
    const {
      updateIsSidebarCollapsed,
      data: { ui: { isSidebarCollapsed, }, },
    } = this.props
    const { swipeWidth, } = this.state

    if ( ( swipeWidth > 100 || isFlick ) && isSidebarCollapsed ) updateIsSidebarCollapsed()

    this.setState( { swipeWidth: 0, } )
  }

  render () {
    const {
      location,
      data: { browser, ui = {}, auth = {}, loading, },
    } = this.props
    const { isSidebarCollapsed, } = ui
    const { isAuthenticated, } = auth
    const { swipeWidth, } = this.state

    const currentKey = location.pathname.split( '/' ).slice( 1, 2 ) || '/'
    const offset = browser > 1200 || swipeWidth || !isSidebarCollapsed ? '12rem' : '4rem'

    return loading
      ? <LoggingIn />
      : (
        <Layout parent>
          {renderRoutes( navMenu, { swipeWidth, } )}
          <Layout>
            {true && <SwipeMask onSwiped={ this.onSwiped } onSwipingRight={ this.onSwipingRight } />}
            {renderRoutes( actionMenu )}
            <TransitionGroup component={ Layout.Content }>
              <GlobalToast offset={ offset } />
              <CSSTransition classNames={ fadeTransition } key={ currentKey } timeout={ 400 }>
                <ContentSwitch isAuthenticated={ isAuthenticated } location={ location } />
              </CSSTransition>
            </TransitionGroup>
          </Layout>
        </Layout>
      )
  }
}
