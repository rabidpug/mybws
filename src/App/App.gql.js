import { compose, graphql, } from 'react-apollo';

import gql from 'graphql-tag';

const gqlApp = compose(
  graphql( gql`
    query LocalAuth {
      auth @client {
        isAuthenticated
      }
      browser @client
      ui @client {
        isSidebarCollapsed
      }
    }
  ` ),
  graphql( gql`
      mutation UpdateOnline {
        updateIsOnline @client
      }
    `,
           { name: 'updateIsOnline', } ),
  graphql( gql`
      mutation UpdateSidebar {
        updateIsSidebarCollapsed @client
      }
    `,
           { name: 'updateIsSidebarCollapsed', } ),
  graphql( gql`
      mutation UpdateBrowser {
        updateBrowser @client
      }
    `,
           { name: 'updateBrowser', } ),
  graphql( gql`
      mutation UpdateLogin($JWT: String, $refreshToken: String) {
        login(JWT: $JWT, refreshToken: $refreshToken) @client
      }
    `,
           { name: 'login', } )
);

export default gqlApp;
