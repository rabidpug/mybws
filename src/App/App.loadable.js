import Loadable from 'react-loadable'
import LoadableLoading from 'Common/helpers/LoadableLoading'
const App = Loadable( {
  delay   : 300,
  loader  : () => import( /*webpackChunkName: "App" */ `./App.container` ),
  loading : LoadableLoading,
  timeout : 10000,
} )

export default App
