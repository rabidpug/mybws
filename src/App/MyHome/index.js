export { default, } from './MyHome.component'
export { default as MyHomeLoadable, } from './MyHome.loadable'
