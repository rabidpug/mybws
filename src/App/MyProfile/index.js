export { default, } from './MyProfile.container'
export { default as MyProfileLoadable, } from './MyProfile.loadable'
