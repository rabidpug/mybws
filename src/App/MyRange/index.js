export { default, } from './MyRange.container'
export { default as MyRangeLoadable, } from './MyRange.loadable'
