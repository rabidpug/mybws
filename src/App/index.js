export { default, } from './App.container'
export { default as App, } from './App.loadable'
