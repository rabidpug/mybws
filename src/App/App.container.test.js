import App from './App.container'
import React from 'react'
import { shallow, } from 'enzyme'
import toJson from 'enzyme-to-json'

describe( 'App Container', () => {
  const AppContainer = () => shallow( <App /> )

  //======== TEST SHALLOW RENDERING ========//
  // it( 'has a TouchableOpacity classname', () => {
  //   expect( appButton().find( 'TouchableOpacity' ).length ).toEqual( 1 );
  // } );

  // it( 'has ActivityIndicator when state loading true', () => {
  //   const wrapper = appButton().setState( { loading: true, } );

  //   expect( wrapper.find( 'ActivityIndicator' ).length ).toEqual( 1 );
  // } );

  // it( 'has no ActivityIndicator when state loading false', () => {
  //   const wrapper = appButton().setState( { loading: false, } );

  //   expect( wrapper.find( 'ActivityIndicator' ).length ).toEqual( 0 );
  // } );

  // it( 'has no Text when state loading false', () => {
  //   const wrapper = appButton().setState( { loading: false, } );

  //   expect( wrapper.find( 'Text' ).length ).toEqual( 1 );
  // } );

  //======== TEST SNAPSHOT =========//
  it( 'should match the snapshot', () => {
    expect( toJson( AppContainer() ) ).toMatchSnapshot()
  } )

  // it( 'snapshot when state loading is true', () => {
  //   const wrapper = appButton().setState( { loading: true, } );

  //   expect( toJson( wrapper ) ).toMatchSnapshot();
  // } );
} )
