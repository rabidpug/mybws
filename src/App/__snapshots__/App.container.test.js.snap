// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Container should match the snapshot 1`] = `
<Route>
  <Component />
</Route>
`;
